be/src/olap/compaction.cpp

```diff
@@ -881,22 +881,30 @@ Status CompactionMixin::modify_rowsets() {
         _tablet->tablet_schema()->cluster_key_idxes().empty()) {
         Version version = tablet()->max_version();
         DeleteBitmap output_rowset_delete_bitmap(_tablet->tablet_id());
-        std::set<RowLocation> missed_rows;
-        std::map<RowsetSharedPtr, std::list<std::pair<RowLocation, RowLocation>>> location_map;
+        std::unique_ptr<RowLocationSet> missed_rows;
+        if ((config::enable_missing_rows_correctness_check ||
```

Comment:
If the binary is debug version, we should enable the check to discover issues in our test pipeline
```
#ifndef NDEBUG
        config::enable_missing_rows_correctness_check = true
#endif
```

Reply:
added in be.conf